colcon_notification/event_handler/status.py

```diff
@@ -53,7 +53,9 @@ def __init__(self):  # noqa: D107
         self._ended = {}
 
         # pattern to match progress indicator in e.g. make
-        self._progress_pattern = re.compile(r'^\[(  \d| \d\d|1\d\d)%\] ')
+        self._progress_pattern_make = re.compile(r'^\[(  \d| \d\d|1\d\d)%\] ')
+        # pattern to match progress indicator in ninja ([X/Y])
+        self._progress_pattern_ninja = re.compile(r'^\[([0-9]+)\/([0-9]+)\] ')
```

Comment:
Can we generalize these variables? Maybe `_progress_pattern_percent` and `_progress_pattern_steps`? If/when we extend this for other tools we'll be able to simply update the existing patterns.